kategory-core/src/main/kotlin/kategory/data/State.kt

```diff
@@ -1,72 +1,125 @@
 package kategory
 
 /**
- * Convenience object to make working with `StateT<IdHK, S, A>` more elegant.
+ * Alias that represents stateful computation of the form `(S) -> Tuple2<S, A>`.
  */
-object State {
+typealias StateFun<S, A> = StateTFun<IdHK, S, A>
 
-    /**
-     * Construct a `StateT<IdHK, S, A>` from a function [run] `(S) -> Tuple2<S, A>`.
-     */
-    operator fun <S, A> invoke(run: (S) -> Tuple2<S, A>): StateT<IdHK, S, A> = StateT(Id(run.andThen { Id(it) }))
+/**
+ * Alias that represents wrapped stateful computation in context `Id`.
+ */
+typealias StateFunKind<S, A> = StateTFunKind<IdHK, S, A>
 
-    /**
-     * Lift a value of type `A` into `StateT<IdHK, S, A>`.
-     */
-    fun <S, A> lift(a: A): StateT<IdHK, S, A> = StateT.lift(Id.monad(), Id.pure(a))
+/**
+ * Alias for StateHK
+ */
+typealias StateHK = StateTHK
+
+/**
+ * Alias for StateKind
+ */
+typealias StateKind<S, A> = StateTKind<IdHK, S, A>
+
+/**
+ * Alias to partially apply type parameters [S] to [State]
+ */
+typealias StateKindPartial<S> = StateTKindPartial<IdHK, S>
+
+/**
+ * `State<S, A>` is a stateful computation that yields a value of type `A`.
+ *
+ * @param S the state we are preforming computation upon.
+ * @param A current value of computation.
+ */
+typealias State<S, A> = StateT<IdHK, S, A>
+
+/**
+ * Constructor for State.
+ * State<S, A> is an alias for IndexedStateT<IdHK, S, S, A>
+ *
+ * @param run the stateful function to wrap with [State].
+ */
+@Suppress("FunctionName")
+fun <S, A> State(run: (S) -> Tuple2<S, A>): State<S, A> = StateT(Id(run.andThen { Id(it) }))
+
+/**
+ * Syntax for constructing a `StateT<IdHK, S, A>` from a function `(S) -> Tuple2<S, A>`
+ */
+fun <S, A> StateFun<S, A>.toState(): State<S, A> = State(this)
+
+/**
+ * Syntax for constructing a `StateT<IdHK, S, A>` from a function `(S) -> Tuple2<S, A>`
+ */
+fun <S, A> StateFunKind<S, A>.toState(): State<S, A> = State(this)
+
+/**
+ * Alias for [StateT.run] `StateT<IdHK, S, A>`
+ *
+ * @param initial state to start stateful computation.
+ */
+fun <S, A> StateT<IdHK, S, A>.run(initial: S): Tuple2<S, A> = run(initial, Id.monad()).value()
+
+/**
+ * Alias for [StateT.runA] `StateT<IdHK, S, A>`
+ *
+ * @param initial state to start stateful computation.
+ */
+fun <S, A> StateT<IdHK, S, A>.runA(initial: S): A = run(initial).b
+
+/**
+ * Alias for [StateT.runS] `StateT<IdHK, S, A>`
+ *
+ * @param initial state to start stateful computation.
+ */
+fun <S, A> StateT<IdHK, S, A>.runS(initial: S): S = run(initial).a
+
+/**
+ * Alias for StateId to make working with `StateT<IdHK, S, A>` more elegant.
+ */
+@Suppress("FunctionName")
+fun State() = StateApi
```

Comment:
can this be declared as a `val` so we are able to do `State.monad()` vs `State().monad()`?

Reply:
Sadly no. `object State` and `val State` both conflict with the `typealias`.

Reply:
👍